lumicks/pylake/fitting/profile_likelihood.py

```diff
@@ -207,9 +214,12 @@ def scan_dir_optimisation(
         current_step_size, p_next = scan_config.step_function(
             chi2_last, p_next, current_step_size, step_sign
         )
-        p_next = fit_function(
-            p_next, scan_config.lower_bounds, scan_config.upper_bounds, scan_config.fitted
-        )
+        try:
+            p_next = fit_function(
+                p_next, scan_config.lower_bounds, scan_config.upper_bounds, scan_config.fitted
+            )
+        except (ValueError, RuntimeError):
+            break
```

Comment:
The way I deal with them is as follows: `ValueError` happens when the model receives an incorrect parameter value (this can be from the user or the optimization algorithm). `RuntimeError` happens if the model encounters something strange that results in an invalid simulation result (but whose inputs are not simply checkable at the door on the model function). This one gets thrown by the likelihood function if the model returned nonsense.

The second question is a good one. My reasoning was as follows: I don't know for certain whether later steps might be fine, but they are unlikely to be fine unless it was a very local numerical phenomenon. Continuing trying could become costly. Reason being that at this stage, we're talking about full optimization cycles, unlike the stepping which is just a single model evaluation. I figured that if we run into something like that, we should give control back to the user. In practice, what this would mean is that the algorithm won't find (or report) a confidence bound on one side, which reveals an issue.

Do you think it would be better to warn and try to struggle on even though it might take a really long time?

Reply:
OK, I agree that immediately stopping and giving back the control to the user upon an issue with the optimization makes sense. I should have read the Pylake documentation to better understand the recommended usage of this method 😅! The information provided there should be sufficient to understand that something went wrong. Still, it might be nice to add a print message if `verbose` is `True` and the profiling is preliminary aborted, so that the user immediately is informed about the issue, if wanted.

Reply:
Yep, that's a good point. I actually think it should go even a little further and just be a full warning. This case should be exceedingly rare, and even if the user doesn't have verbose set, I think it's wise to alert them about it. I've also added a test which tests explicitly whether this warning fires.

Reply:
I like the warning!